pinecone/client_test.go

```diff
@@ -285,6 +327,34 @@ func (ts *ClientTests) TestCreatePodIndex() {
 	require.Equal(ts.T(), name, idx.Name, "Index name does not match")
 }
 
+func (ts *ClientTests) TestCreatePodIndexInvalidDimension() {
```

Comment:
I wasn't sure how deep to go with these integration tests that intentionally cause errors. I can add more if they seem useful.